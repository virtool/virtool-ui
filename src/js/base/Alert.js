import { get } from "lodash-es";
import React from "react";
import styled from "styled-components";
import { getColor, getFontWeight } from "../app/theme";
import { Icon } from "./Icon";

export const getAlertBackgroundColor = ({ color, theme }) =>
    get(theme, ["color", `${color}Lightest`], theme.color.greyLightest);

export const getAlertTextColor = ({ color, theme }) => get(theme, ["color", `${color}Dark`], theme.color.greyDarkest);

const AlertInner = styled.div`
    align-items: ${props => (props.level ? "center" : "normal")};
    border-left: 10px solid ${getColor};
    display: ${props => (props.block ? "block" : "flex")};
    padding: 15px;
`;

const StyledAlert = styled.div`
    background-color: ${getAlertBackgroundColor};
    border-radius: ${props => props.theme.borderRadius.md};
    border: none;
    box-shadow: ${props => props.theme.boxShadow.sm};
    color: ${getAlertTextColor};
    font-weight: ${getFontWeight("thick")};
    margin-bottom: 16px;
    overflow: hidden;

    i.fas:first-child {
        color: currentColor;
        padding-right: 5px;
    }
`;

export const _Alert = ({ block, children, className, color, icon, level }) => (
    <StyledAlert color={color}>
        <AlertInner className={className} block={block} color={color} level={level}>
            {icon ? <Icon name={icon} /> : null}
            {children}
        </AlertInner>
    </StyledAlert>
);

export const Alert = styled(_Alert)``;
