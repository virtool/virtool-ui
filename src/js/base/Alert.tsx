import React from "react";
import styled from "styled-components";
import { AlertInner } from "./AlertInner";
import { AlertOuter } from "./AlertOuter";
import { Icon } from "./Icon";

type AlertProps = {
    block?: boolean;
    children: React.ReactNode;
    className?: string;
    color?: string;
    icon?: string;
    level?: boolean;
};

export const Alert = styled(({ block, children, className, color, icon, level }: AlertProps) => (
    <AlertOuter color={color}>
        <AlertInner className={className} block={block} color={color} level={level}>
            {icon ? <Icon name={icon} /> : null}
            {children}
        </AlertInner>
    </AlertOuter>
))``;

Alert.displayName = "Alert";
