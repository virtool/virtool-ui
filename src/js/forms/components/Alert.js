import React from "react";
import styled from "styled-components";
import { Alert, Icon } from "../../base";
import { getColor } from "../../app/theme";

const StyledAlert = styled(Alert)`
    display: flex;
    align-items: center;

    button {
        margin-left: 10px;
        color: ${props => getColor({ color: "greyDark", theme: props.theme })};
        :hover {
            color: ${props => getColor({ color: "greyDarkest", theme: props.theme })};
        }
        :last-child {
            margin-left: auto;
            font-size: 16px;
        }
    }
`;

export const RestoredAlert = ({ onClose, resetForm }) => {
    const onUndoRestore = () => {
        resetForm();
        onClose();
    };

    return (
        <StyledAlert>
            <span>Resumed editing draft sample.</span>
            <Icon aria-label="undo restore" name="undo" color="grey" onClick={onUndoRestore} />
            <Icon aria-label="close" name="times" color="grey" onClick={onClose} />
        </StyledAlert>
    );
};
