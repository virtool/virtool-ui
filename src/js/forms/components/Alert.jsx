import React from "react";
import styled from "styled-components";
import { getColor } from "../../app/theme";
import { Alert, Icon } from "../../base";

const StyledRestoredAlert = styled(Alert)`
    display: flex;
    align-items: center;

    button {
        margin-left: 10px;
        color: ${props => getColor({ color: "greyDark", theme: props.theme })};

        :hover {
            color: ${props => getColor({ color: "greyDarkest", theme: props.theme })};
        }
        :last-child {
            margin-left: auto;
            font-size: 16px;
        }
    }
`;

export const RestoredAlert = ({ onClose, resetForm }) => {
    const onUndoRestore = () => {
        resetForm();
        onClose();
    };

    return (
        <StyledRestoredAlert>
            <span>Resumed editing draft sample.</span>
            <Icon aria-label="undo restore" name="undo" color="grey" onClick={onUndoRestore} />
            <Icon aria-label="close" name="times" color="grey" onClick={onClose} />
        </StyledRestoredAlert>
    );
};
