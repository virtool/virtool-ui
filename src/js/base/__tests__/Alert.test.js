import { Alert } from "../Alert";

describe("<Alert />", () => {
    it.each(["blue", "green", "red"])("should render when [color=%p]", color => {
        const wrapper = shallow(<Alert color={color} />);
        expect(wrapper).toMatchSnapshot();
    });
});
