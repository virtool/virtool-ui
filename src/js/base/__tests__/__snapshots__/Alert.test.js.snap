// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render when [color="blue"] 1`] = `
<_Alert
  className=""
  color="blue"
/>
`;

exports[`<Alert /> should render when [color="green"] 1`] = `
<_Alert
  className=""
  color="green"
/>
`;

exports[`<Alert /> should render when [color="red"] 1`] = `
<_Alert
  className=""
  color="red"
/>
`;
